File: tools/conf/output/ts/BgConfig.ts
class BgConfig {
    private static _dic:Dictionary<BgConfig> = new Dictionary<BgConfig>();
    private static _arr:Array<BgConfig> = new Array<BgConfig>();

	 public Id :number ;
     public Type :number ;
     public Imgs : Array<number>  =new Array<number>();
     public Floor :number ;
     public Particle : Array<number>  =new Array<number>();
     public BgMusic :number ;
    
    

    
    public constructor(){

    }


    public static get Dic():Dictionary<BgConfig>{
        return  BgConfig._dic;
    }
    public static get Arr():Array<BgConfig>{
        return BgConfig._arr;
    }
    public static parse(json:any[]){

        for(var i = 0; i < json.length; i++){
            let config:BgConfig = <BgConfig>json[i];
            BgConfig._arr.push(config);
            BgConfig._dic.add(config.Id,config);
        }
    }
}

File: server/src/game/tools/conf/output/go/BgConfig.go
package conf

import (
	//"bufio"
	"encoding/json"
	"fmt"
	//"io"
	"io/ioutil"
	"os"
	//"strings"
)

type BgConfigElem struct {
	Id int
	Type int
	Imgs []int
	Floor int
	Particle []int
	BgMusic int
	Des string
	
}

type BgConfig0 struct {
	Dic map[int]*BgConfigElem
	Arr []*BgConfigElem
}

var (
	BgConfig BgConfig0
)

func loadBgConfig(path string) {
	file, err := os.Open(path)
	defer func() {
		if file != nil {
			file.Close()
		}
	}()
	if err != nil {
		panic(fmt.Sprint("load config error, path=", path))
	}

	BgConfig.Dic = make(map[int]*BgConfigElem)
	BgConfig.Arr = make([]*BgConfigElem, 0, 32)

	bt,err  :=  ioutil.ReadAll(file)
	json.Unmarshal(bt, &BgConfig.Arr)
	for _,conf := range BgConfig.Arr{
		BgConfig.Dic[conf.Id] = conf
	}
	
}
